edx/edx-enterprise
# ENT-97: Email learners on enrollment

**Description:**

This pull request adds a feature that enables platform owners to email new learners upon enrollment in a course, using either a built-in template with approved text, or by setting up a custom, site-specific, set of templates that can be used to render messages.

**JIRA:** [ENT-97](https://openedx.atlassian.net/browse/ENT-97)

**Dependencies:** Depends on an incoming PR from @e-kolpakov; the ticket acceptance criteria specify that learners can be notified when being enrolled in a program. However, program enrollment does not exist as of yet.

**Merge deadline:** 30 January 2017

**Sandbox Details:** TBD (being provisioned)

**Testing instructions:**

1. In the Manage Learners page in a devstack (where emails are logged to the console), add a learner to a course using an arbitrary email address.
    - Observe in the console that an email has been "sent" using both the standard plaintext template and the standard HTML template - email clients will choose the appropriate one to render.
2. Set up an `EnrollmentNotificationEmailTemplate` object in the Enterprise admin; be sure to fill in templates for both the plaintext and HTML templates, as well as for the subject line.
3. Again, add a learner to a course using an arbitrary email address.
    - Observe in the console that the email was "sent", and that your custom templates were used to render the message, rather than the standard templates.

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:**

1. More tests are still incoming; this is not entirely complete yet.
2. A flag needs to be added to the Enterprise admin to make notification optional
3. Because program enrollment doesn't exist yet, notification on program enrollment isn't implemented yet.
